service-search-nls/src/main/java/fi/nls/oskari/search/channel/OgcApiSearchChannel.java

```diff
@@ -0,0 +1,202 @@
+package fi.nls.oskari.search.channel;
+
+import fi.mml.portti.service.search.ChannelSearchResult;
+import fi.mml.portti.service.search.SearchCriteria;
+import fi.mml.portti.service.search.SearchResultItem;
+import fi.nls.oskari.annotation.Oskari;
+import fi.nls.oskari.log.LogFactory;
+import fi.nls.oskari.log.Logger;
+import fi.nls.oskari.search.channel.SearchChannel;
+import fi.nls.oskari.util.IOHelper;
+import fi.nls.oskari.util.PropertyUtil;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UnsupportedEncodingException;
+import java.net.URLDecoder;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.*;
+
+
+@Oskari(OgcApiSearchChannel.ID)
+public class OgcApiSearchChannel extends SearchChannel {
+
+
+    public static final String ID = "OGC_API_SEARCH_CHANNEL";
+
+    //Local configs
+    /**
+     * One can use property search.channel.OGC_API_SEARCH_CHANNEL.apikey but if really needed apikey can go here.
+     */
+    private static String apiKey = null;
+    private static String searchAddress = "https://avoin-paikkatieto.maanmittauslaitos.fi/geographic-names/features/v1/collections/placenames/items?spelling_case_insensitive=";
+    private static String additionalPlaceTypeInfoBaseUrl = "https://beta-paikkatieto.maanmittauslaitos.fi/catalogue/v1/codelistgroups/geographic-names/codelists/";
+    private static String[] localAdditionalInfoNames = {"placeType", "municipality"};
```

Comment:
The array isn't helping here. You are not looping through it ever but always referencing `localAdditionalInfoNames[0]` or `localAdditionalInfoNames[1]` making it harder to read than it needs to be. Separate these in two constants that have names better suited for the purpose.